server.go

```diff
@@ -824,7 +824,7 @@ func (s *Server) Stop() {
 func (s *Server) GracefulStop() {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	if s.drain == true || s.conns == nil {
+	if s.conns == nil {
```

Comment:
We should avoid calling ServerTransport.Drain() multiple times in Line#836. We should move line#830 below line#837 and add the if branch between line#834 and line#835 like:

```
if !s.drain {
  for c:= range s.conns {
      c.(transport.ServerTransport).Drain()
  }
  s.drain = true
}
```